### PR TITLE
Update node version

### PR DESCRIPTION
**Pull request by bdovaz**
Fixes https://github.com/Dinomite-Studios/unity-azure-pipelines-tasks/issues/258

I have taken the opportunity to change the format of the versions so that it adapts to the schema (int instead of string): https://github.com/microsoft/azure-pipelines-task-lib/blob/master/tasks.schema.json

### CODE REVIEW

**Comment by bdovaz** [comment]
@FejZa I don't know what's wrong because I can't see the pipeline, it gives me a 401 error....

**Comment by FejZa** [comment]
Hi @bdovaz, thank you! Here's the build error:

`node_modules/@types/node/module.d.ts(106,13): error TS2386: Overload signatures must all be optional or required.`

I'll look into making build logs accessible for everyone.

**Comment by bdovaz** [comment]
> I'll look into making build logs accessible for everyone.

I would appreciate it because otherwise I have no way to test it on my own without bothering you.

**Comment by bdovaz** [comment]
> I believe you'll have to update the `@types/node` dependency here as well.

@FejZa I just updated the dependency but it keeps failing and I don't know what it is 😅 I don't have experience in TS

File: Tasks/UnityActivateLicense/UnityActivateLicenseV1/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
   "author": "Dinomite Studios",
   "license": "MIT",
   "dependencies": {
-    "@types/node": "^18.0.0",
+    "@types/node": "^20.0.0",
```
Comment (TimO143) [inline Tasks/UnityActivateLicense/UnityActivateLicenseV1/package.json:20]: I am just a bystanding with just a little knowledge as the next person wanting to find a fix for this :P , but shouldn't this version be "@types/node": "^20.3.1"
according to the migration documentation?
https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/migrateNode20.md

**Comment by FejZa** [comment]
Hey @bdovaz, the node update definitely worked, since the License and Build Unity Tasks are now compiling correctly. But the pipeline fails at the UnityCMD task which seems to have an actual build error now by the TypeScript compiler:

`unity-cmd.ts(34,34): error TS2872: This kind of expression is always truthy.`

Btw. you can always test locally by running `npm run build` in the respective task's directory.

**Comment by bdovaz** [comment]
> Hey @bdovaz, the node update definitely worked, since the License and Build Unity Tasks are now compiling correctly. But the pipeline fails at the UnityCMD task which seems to have an actual build error now by the TypeScript compiler:
> 
> `unity-cmd.ts(34,34): error TS2872: This kind of expression is always truthy.`
> 
> Btw. you can always test locally by running `npm run build` in the respective task's directory.

That one I have corrected it but in local I see others.... I don't know if it has to do that I'm compiling it with Node v22 which is the one I have installed on my computer, I'll wait to see what you tell me about the result of pipelines if it fails.

```
node_modules/@types/node/stream/web.d.ts:462:13 - error TS2502: 'ReadableByteStreamController' is referenced directly or indirectly in its own type annotation.

462         var ReadableByteStreamController: typeof globalThis extends
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/node/stream/web.d.ts:471:13 - error TS2502: 'ReadableStreamBYOBReader' is referenced directly or indirectly in its own type annotation.

471         var ReadableStreamBYOBReader: typeof globalThis extends { onmessage: any; ReadableStreamBYOBReader: infer T }
                ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@types/node/stream/web.d.ts:476:13 - error TS2502: 'ReadableStreamBYOBRequest' is referenced directly or indirectly in its own type annotation.

476         var ReadableStreamBYOBRequest: typeof globalThis extends { onmessage: any; ReadableStreamBYOBRequest: infer T }
                ~~~~~~~~~~~~~~~~~~~~~~~~~
```

I have taken the opportunity to correct some warnings that in theory were easy to fix with sonar VS Code plugin hints

**Comment by bdovaz** [comment]
@FejZa well, it seems that now it has compiled everything in pipelines, check the changes of warnings that I have done in case I have created some problem so you could merge it.

**Review by FejZa — APPROVED** [review]

**Pull request merged**